Core/DotLanches.Application/UseCases/PagamentoUseCases.cs

```diff
@@ -6,7 +6,7 @@ namespace DotLanches.Application.UseCases
 {
     public static class PagamentoUseCases
     {
-        public static async Task<Pagamento> ProcessPagamento(int idPedido, IPedidoGateway pedidoGateway, IPagamentoGateway pagamentoGateway, ICheckout checkout)
+        public static async Task<string> ProcessPagamento(int idPedido, IPedidoGateway pedidoGateway, IPagamentoGateway pagamentoGateway, ICheckout checkout)
```

Comment:
Sugestão: deixar os casos de uso mais próximos da linguagem de negócio, indicando o que o sistema faz:

```suggestion
        public static async Task<string> RequestQrCodeForPedido(int idPedido, IPedidoGateway pedidoGateway, IPagamentoGateway pagamentoGateway, ICheckout checkout)
```